x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialUtils.java

```diff
@@ -0,0 +1,58 @@
+/*
```

Comment:
This should not be in the `ql` module as it is only used in `esql`. In addition the class name is not useful, since it is specific to coordinate encoding. Consider renaming and moving to the `esql` plugin.

Reply:
I moved this code to the ESQL module, but then server tests failed on classloader errors, so I've moved it back until we figure out what is going on with classloading. All other changes like renaming the class, and making it an enum in preparation for Cartesian support are retained. The only unexpected requirement is that this class must be in the QL module for some reason.